src/guarneri/instrument.py

```diff
@@ -50,49 +49,127 @@ class Instrument:
 
     """
 
-    devices: list
-    registry: Registry
-    beamline_name: str = ""
-    hardware_is_present: bool | None = None
+    devices: Registry
 
     def __init__(self, device_classes: Mapping, registry: Registry | None = None):
-        self.devices = []
+        self.unconnected_devices = []
         if registry is None:
             registry = Registry(auto_register=False, use_typhos=False)
-        self.registry = registry
+        self.devices = registry
         self.device_classes = device_classes
 
-    def parse_toml_file(self, fd):
-        """Parse TOML instrument configuration and create devices.
+    def parse_config(self, config_file: Path | str) -> list[dict]:
+        """Parse an instrument configuration file.
+
+        This method can be overridden to implement custom
+        configuration file schema. It should return a sequence of
+        device definitions, similar to:
+
+        .. code-block:: python
+
+            [
+                {
+                   "device_class": "ophyd.motor.EpicsMotor",
+                   "args": (),
+                   "kwargs": {
+                       "name": "my_device",
+                       "prefix": "255idcVME:m1",
+                    },
+                }
+            ]
+
+        *device_class* can be an entry in the
+        ``Instrument.device_classes``, or else an import path that
+        will be loaded dynamically.
+
+        Parameters
+        ==========
+        config_file
+          A file path to read.
+
+        Returns
+        =======
+        device_defns
+          A list of dictionaries, describing the devices to create.
 
-        An open file descriptor
         """
-        config = tomlkit.load(fd)
-        # Set global parameters
-        beamline = config.get("beamline", {})
-        self.beamline_name = beamline.get("name", self.beamline_name)
-        self.hardware_is_present = beamline.get(
-            "hardware_is_present", self.hardware_is_present
-        )
-        # Make devices from config file
-        return self.parse_config(config)
+        config_file = Path(config_file)
+        if config_file.suffix == ".toml":
+            return self.parse_toml_file(config_file)
+        if config_file.suffix == ".yaml":
```

Comment:
It's common for YAML files to use either `.yaml` or `.yml` extensions.  Code should recognize both.

```suggestion
        if config_file.suffix in (".yaml", ".yml"):
```